localstack/services/awslambda/lambda_executors.py

```diff
@@ -1577,7 +1576,21 @@ def get_java_opts(cls):
 
     @classmethod
     def get_host_path_for_path_in_docker(cls, path):
-        return re.sub(r"^%s/(.*)$" % config.dirs.tmp, r"%s/\1" % config.dirs.functions, path)
+        if config.LEGACY_DIRECTORIES:
+            return re.sub(r"^%s/(.*)$" % config.dirs.tmp, r"%s/\1" % config.dirs.functions, path)
+
+        if config.is_in_docker:
+            for mount in inspect_current_container()["Mounts"]:
```

Comment:
This is _awesome_ - no more need to configure `HOST_TMP_FOLDER`. 🥳 This will make things a lot easier to use, and I reckon will drastically reduce the number of support requests we're getting for Lambda function mounting! Love it.

nit: We could cache the result of the volume `Mounts` in the Docker inspect result, to optimize performance. (mounts should not change for the duration of the container's lifetime)